app/models/concerns/dirty_associations.rb

```diff
@@ -14,16 +14,30 @@
 module DirtyAssociations
   extend ActiveSupport::Concern
 
+  included do
+    after_save :reset_dirty_cache_state
+  end
   module ClassMethods
     # usage:
     #   class Model
     #     dirty_has_many_associations :organizations, :locations
+
     def dirty_has_many_associations(*args)
+      @@dirty_associations ||= []
+      @@dirty_associations += args
+      dirty_associations = @@dirty_associations
       extension = Module.new do
         args.each do |association|
           association_ids = association.to_s
           association_ids = association_ids.singularize + '_ids' unless association.to_s.end_with?('_ids')
 
+          define_method "reset_dirty_cache_state" do
```

Comment:
since you now define this in args.each block, the name should be dynamic, something like `define_method "reset_#{association}_dirty_cache_state"`